lib/src/source/hosted.dart

```diff
@@ -1100,3 +1109,91 @@ class _RefAndCache {
   @override
   bool operator ==(Object other) => other is _RefAndCache && other.ref == ref;
 }
+
+const checksumHeaderName = 'x-goog-hash';
+
+bool _hasChecksumHeader(Map<String, String> headers) {
+  return headers.containsKey(checksumHeaderName);
+}
+
+/// Adds a checksum validation "tap" to the response stream and returns a
+/// wrapped Stream object, which should be used to consume the incoming data.
+///
+/// As chunks are received, a CRC32C checksum is updated.
+/// Once the download is completed, the final checksum is compared with
+/// the one present in the checksum response header.
+///
+/// Throws [PackageIntegrityException] if anything is wrong with the checksum
+/// or if there is a checksum mismatch.
+Stream<List<int>> _responseStreamWithChecksumValidationTap(
+    http.StreamedResponse response, String fileName) {
```

Comment:
I don't think you are using the file name here.

Reply:
So, I'm using it as diagnostic information in the "log.fine" call a few lines below.

`log.fine('Computed CRC32C ($crc32c) for $fileName with hosted CRC32C of ($hostedCrc32c)');`

I was not 100% happy with the solution of passing `fileName`, but I couldn't come up with a better way in time so I left it like that.